hypothesis-python/src/hypothesis/strategies/_internal/lazy.py

```diff
@@ -132,27 +132,21 @@ def do_validate(self):
 
     def __repr__(self):
         if self.__representation is None:
-            _args = self.__args
-            _kwargs = self.__kwargs
-            argspec = getfullargspec(self.function)
-            defaults = dict(argspec.kwonlydefaults or {})
-            if argspec.defaults is not None:
-                for name, value in zip(
-                    reversed(argspec.args), reversed(argspec.defaults)
-                ):
-                    defaults[name] = value
-            if len(argspec.args) > 1 or argspec.defaults:
+            sig = signature(self.function)
```

Comment:
To preserve behavior of `getfullargspec`:

`sig = signature(f, follow_wrapped=False)`

Reply:
(there are additional instances below, but I'll hold off on pointing them all out in case I am simply wrong here)